test-suite-kotlin/src/test/kotlin/io/micronaut/microstream/docs/CustomerController.kt

```diff
@@ -0,0 +1,40 @@
+package io.micronaut.microstream.docs
+
+import io.micronaut.http.HttpResponse
+import io.micronaut.http.HttpStatus
+import io.micronaut.http.annotation.Get
+import io.micronaut.http.annotation.Post
+import io.micronaut.http.annotation.Controller
+import io.micronaut.http.annotation.Status
+import io.micronaut.http.annotation.Delete
+import io.micronaut.http.annotation.Body
+import io.micronaut.http.annotation.PathVariable
+import io.micronaut.http.uri.UriBuilder
+import java.util.Optional
+import java.util.UUID
+import javax.validation.Valid
+
+@Controller("/customer")
+internal class CustomerController(private val repository: CustomerRepository) {
+    @Post
+    fun save(@Body customerSave: @Valid CustomerSave): HttpResponse<*> {
```

Comment:
Is putting `@Valid` before the type idiomatic?

Reply:
If I change from: 

```kotlin
@Post
fun save(@Body @Valid customerSave: CustomerSave): HttpResponse<*> {
```

to: 

```kotlin
@Post
    fun save(@Body @Valid customerSave: CustomerSave): HttpResponse<*> {
``` 


I get: 


```
CustomerController.java:17: error: Method defines AOP advice but is declared final. Change the method to be non-final in order for AOP advice to be applied.
    public final io.micronaut.http.HttpResponse<?> save(@org.jetbrains.annotations.NotNull()
                                                   ^

```

Reply:
Weird...  and the validation works on the save method?